src/Gql.js

```diff
@@ -0,0 +1,215 @@
+import _ from 'lodash';
+import {parse, validate, execute} from 'graphql';
+
+import {formatError} from './errors';
+import {makeSchemaFromConfig} from './makeSchemaFrom';
+
+const ONE_DAY = 24 * 3600 * 1000;
+const ENUM_PREFIX = '__ENUM__::';
+// const VAR_PREFIX = '__VAR__::';
+
+// const NO_QUOTES_REGEX = new RegExp(`^(${ENUM_PREFIX}|${VAR_PREFIX})([A-Za-z_]+)$`);
+const ENUM_REGEX = new RegExp(`^${ENUM_PREFIX}([A-Za-z_]+)$`);
+
+function graphql(schemaObj, query, rootValue, contextValue, variableValues) {
+	// we are not using the inbuilt graphql function because it validates
+	// the graphql, which is an expensive operation
+	// return graphql(schema, query, rootValue, contextValue, variableValues);
+	// taken from: https://github.com/graphql/graphql-js/blob/master/src/graphql.js
+
+	// parse
+	let document;
+	try {
+		document = parse(query);
+	}
+	catch (syntaxError) {
+		return Promise.resolve({errors: [syntaxError]});
+	}
+
+	// validate only in developement
+	if (process.env.NODE_ENV !== 'production') {
+		const validationErrors = validate(schemaObj, document);
+		if (validationErrors.length > 0) {
+			return Promise.resolve({errors: validationErrors});
+		}
+	}
+
+	return execute(
+		schemaObj,
+		document,
+		rootValue,
+		contextValue,
+		variableValues,
+	);
+}
+
+function convertObjToGqlArg(obj) {
+	const gqlArg = [];
+	_.forEach(obj, (value, key) => {
+		// eslint-disable-next-line no-use-before-define
+		gqlArg.push(`${key}: ${convertToGqlArg(value)}`);
+	});
+	return `${gqlArg.join(', ')}`;
+}
+
+function convertStrToGqlArg(str) {
+	const [, val] = str.match(ENUM_REGEX) || [];
+	if (val) return val;
+	// const [, prefix, val] = str.match(NO_QUOTES_REGEX) || [];
+	// if (prefix && val) return prefix === VAR_PREFIX ? '$' + val : val;
+
+	return JSON.stringify(str);
+}
+
+function convertToGqlArg(value) {
+	if (value == null) return null;
+
+	if (typeof value === 'number') return String(value);
+	if (typeof value === 'string') return convertStrToGqlArg(value);
+	if (_.isPlainObject(value)) return `{${convertObjToGqlArg(value)}}`;
+
+	return JSON.stringify(value);
+}
+
+class Gql {
+	constructor(opts = {}) {
+		const {schema, pubsub, defaultSchema} = makeSchemaFromConfig(opts);
+
+		this.schema = schema;
+		this.defaultSchema = defaultSchema;
+		this.pubsub = pubsub;
+
+		this.logger = opts.logger || console;
+		this.cache = opts.cache;
+		this.formatError = opts.formatError || formatError;
+	}
+
+	async exec(query, context, {
+		cache: {key: cacheKey, ttl = ONE_DAY} = {},
+		variables = {},
+	} = {}) {
+		if (cacheKey && this.cache) {
+			const cached = await this.cache.get(cacheKey);
+			if (cached) return cached;
+		}
+
+		if (!/^\s*query|mutation|subscription/.test(query) && /^\s*[a-zA-Z0-9]/.test(query)) {
+			query = `query { ${query} }`;
+		}
+
+		const result = await graphql(this.defaultSchema, query, null, context, variables);
```

Comment:
this should be `this.schema`, the selected one via makeSchemaFromConfig options

Reply:
`schema.default` is being passed here. So I used the `defaultSchema`.
https://github.com/smartprix/smartprix/blob/0822159767c6bb312d80c563f9079ff20123b19d/src/graphql.js#L121